src/Tickets/Commerce/Order_Modifiers/Controller.php

```diff
@@ -0,0 +1,178 @@
+<?php
+/**
+ * Event Tickets Order Modifiers Controller.
+ *
+ * @since   TBD
+ * @package TEC\Tickets\Commerce\Order_Modifiers
+ */
+
+declare( strict_types=1 );
+
+namespace TEC\Tickets\Commerce\Order_Modifiers;
+
+use TEC\Common\Contracts\Provider\Controller as Controller_Contract;
+use TEC\Tickets\Commerce\Order_Modifiers\Admin\Order_Modifier_Fee_Metabox;
+use TEC\Tickets\Commerce\Order_Modifiers\Modifiers\Fee;
+use TEC\Tickets\Commerce\Order_Modifiers\Checkout\Gateway\Paypal\Fees as Paypal_Checkout_Fees;
+use TEC\Tickets\Commerce\Order_Modifiers\Checkout\Gateway\Stripe\Fees as Stripe_Checkout_Fees;
+use TEC\Tickets\Commerce\Order_Modifiers\Checkout\Fees as Agnostic_Checkout_Fees;
+use TEC\Tickets\Commerce\Order_Modifiers\API\Coupons;
+use TEC\Tickets\Commerce\Order_Modifiers\API\Fees;
+use TEC\Tickets\Commerce\Order_Modifiers\Modifiers\Coupon;
+use TEC\Tickets\Commerce\Order_Modifiers\Traits\Valid_Types;
+use TEC\Tickets\Commerce\Order_Modifiers\Modifiers\Modifier_Strategy_Interface;
+use InvalidArgumentException;
+use TEC\Tickets\Commerce\Order_Modifiers\Admin\Editor;
+use TEC\Common\StellarWP\Assets\Config;
+use Tribe__Tickets__Main as Tickets_Plugin;
+
+/**
+ * Main Order Modifiers Controller.
+ *
+ * @since TBD
+ */
+final class Controller extends Controller_Contract {
+
+	use Valid_Types;
+
+	/**
+	 * Un-registers the Controller by unsubscribing from WordPress hooks.
+	 *
+	 * Bound implementations should not be removed in this method!
+	 *
+	 * @since TBD
+	 *
+	 * @return void Filters and actions hooks added by the controller are be removed.
+	 */
+	public function unregister(): void {
+		$this->container->get( Paypal_Checkout_Fees::class )->unregister();
+		$this->container->get( Stripe_Checkout_Fees::class )->unregister();
+		$this->container->get( Agnostic_Checkout_Fees::class )->unregister();
+		$this->container->get( Tables::class )->unregister();
+		$this->container->get( Editor::class )->unregister();
+
+		if ( is_admin() ) {
+			$this->container->get( Modifier_Admin_Handler::class )->unregister();
+			$this->container->get( Order_Modifier_Fee_Metabox::class )->unregister();
+		}
+
+		if ( $this->container->isBound( Coupons::class ) ) {
+			$this->container->get( Coupons::class )->unregister();
+			return;
+		}
+
+		remove_filter( 'tec_tickets_commerce_order_modifiers', [ $this, 'filter_out_coupons' ] );
+		remove_filter( 'tec_tickets_commerce_order_modifier_types', [ $this, 'filter_out_coupons' ] );
+	}
+
+	/**
+	 * Registers the service provider bindings.
+	 *
+	 * @return void The method does not return any value.
+	 */
+	public function do_register(): void {
```

Comment:
I don't want someone to take the time to change them all, but is there a reason we put `unregister` _before_ `do_register` in all of these?

I know the order doesn't matter, but readability-wise I want to see what's registered before I see what's unregistered 🤷🏼

Reply:
I mostly agree. I think though thats not really a concern and we should allow people to write methods in whatever order they like. Otherwise we are becoming stricter than the more strict coding standards.

As you said it doesn't matter and what if another person comes and say for me readability wise is to see first the callbacks ?